src/main/java/mops/businesslogic/FileServiceImpl.java

```
@@ -0,0 +1,212 @@
+        // Only true if user is not the owner and has no delete permission
+        if (!isOwner && !userPermission.isDelete()) {
+            log.error("User {} tried to delete file with ID {} without permission.",
+                    account.getName(),
+                    fileId
+            );
+            throw new DeleteAccessPermissionException("No delete permission");
+        }
+
+        try {
+            fileInfoService.deleteFileInfo(fileId);
+            fileRepository.deleteFile(fileId);
+        } catch (MopsException e) {
+            TransactionAspectSupport.currentTransactionStatus().setRollbackOnly();
+            log.error("File with ID {} error on retrieving. Error: {}",
+                    fileId,
+                    e.getMessage()
+            );
+            throw new MopsException("Error while deleting", e);
+        }
+        return directoryService.getDirectory(fileInfo.getDirectoryId());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @SuppressWarnings({ "PMD.LawOfDemeter", "PMD.DataflowAnomalyAnalysis" })
+    public FileInfo getFileInfo(Account account, long fileId) throws MopsException {
+        FileInfo fileInfo;
+        try {
+            fileInfo = fileInfoService.fetchFileInfo(fileId);
+        } catch (MopsException e) {
+            log.error("User {} tried to retrieved non existing file with ID {}. Error: {}",
+                    account.getName(),
+                    fileId,
+                    e.getMessage()
+            );
+            throw new FileNotFoundException("File not found", e);
+        }
+        UserPermission userPermission = directoryService.getPermissionsOfUser(account, fileInfo.getDirectoryId());
+
+        if (!userPermission.isRead()) {
+            log.error("User {} tried to read file {} without permission.",
+                    account.getName(),
+                    fileId,
+                    new ReadAccessPermissionException("No read permission")
```

you need to throw this exception instead of giving it to the logger.